addons/sale_project/models/sale_order.py

```diff
@@ -201,7 +201,7 @@ def action_view_project_ids(self):
             return {'type': 'ir.actions.act_window_close'}
 
         sorted_line = self.order_line.sorted('sequence')
-        default_sale_line = next(sol for sol in sorted_line if sol.product_id.detailed_type == 'service')
+        default_sale_line = sorted_line.search([('product_id.detailed_type', '=', 'service')], limit=1)
```

Comment:
search will always search over all record of the database

what you would want to search on the current recordset would be things like:

```suggestion
        default_sale_line = sorted_line.search([('product_id.detailed_type', '=', 'service'), ('id', 'in', sorted_line.ids)], limit=1)
        # or
        default_sale_line = sorted_line.filtered_domain([('product_id.detailed_type', '=', 'service')])[:1]
```
but you could also just keep the next and have a second parameter for fallback, eg:
```suggestion
        default_sale_line = next(sol for sol in sorted_line if sol.product_id.detailed_type == 'service', None)
```

Reply:
I initially thought it was the depreciation of detailed_type that caused the issue not to occur in 17.4 but it turns out we actually do have a difference there that looks a lot like what you're suggesting https://github.com/odoo/odoo/blob/saas-17.4/addons/sale_project/models/sale_order.py#L228-L231

I'll follow their implementation then, thanks for the help :+1: